Mu2eG4/src/toggleProcesses.cc

```diff
@@ -213,6 +214,29 @@ namespace mu2e{
       pmanager->AddContinuousProcess(rmp);
       pmanager->SetProcessOrderingToLast(rmp, idxAlongStep);
     }
+
+
+    if(phys.gammaFilterMinDaughterEnergy() > 0.) {
+      // special process to filter gamma conversions
+      if (diagLevel>=0) {
+        G4cout << __func__
+               << " adding Mu2eGammaDaughterProcess to relevant G4ParticleTable particles with E_min = "
+               << phys.gammaFilterMinDaughterEnergy() << G4endl;
+      }
+      Mu2eGammaDaughterProcess* gdp = new Mu2eGammaDaughterProcess();
```

Comment:
It looks like this constructor creates an incomplete object, requiring 'set' function to be usable.  A better design is to pass all required parameters to the constructor, so it is complete on construction.  Then you can get rid of the 'set' functions.

Reply:
I agree these should be set upon construction. I added these to the constructor as suggested.